packages/atomic/src/index.ts

```diff
@@ -6,12 +6,6 @@ export {
   initializeBindings,
   Bindings,
   MissingInterfaceParentError,
-  BindStateToController,
-  InitializeBindings,
-  InitializableComponent,
 } from './utils/initialization-utils';
 
-export {
-  ResultContext,
-  resultContext,
-} from './components/result-template-components/result-template-decorators';
+export {resultContext} from './components/result-template-components/result-template-decorators';
```

Comment:
I should’ve investigated this more before exporting the decorators. It kind of worked because of the collection output recompiling the Stencil output, but it threw a some errors. Now that the collection output is deleted, the decorators aren’t compatible. They are typescript decorators, compiled into js decorators, which can’t override the component methods such as connected callback. The utils method still work, bellow are linked examples of a custom component and a custom result template component. Not sure if exposing a typescript file with typescript decorators is a good idea, I haven’t many example of packages like this online. We might have to improve upon this later…

- https://github.com/ThibodeauJF/stencil-app-atomic/tree/master/src/components/custom-pager
- https://github.com/ThibodeauJF/stencil-app-atomic/blob/master/src/components/custom-title/custom-title.tsx

Reply:
> Not sure if exposing a typescript file with typescript decorators is a good idea

Exporting only javascript is recommended because it avoids non-typescript consumers (e.g. babel, pure js etc.) needing to transpile our code. However, since decorators are not natively supported in js, then it seems like we have two options. Either:
- We state in our docs that transpiling with tsc or babel is needed.
- We offer a non-decorator approach.

Reply:
In a Stencil dev project, we can expect typescript to be used. My current preferred solution is using the non decorator approaches (initializeBindings, resultContext methods) and in the CLI setup project, which is written in typescript, having an available decorator. I think it's important that the user can see and edit the decorator, as it has a lot of side effects.

Reply:
> and in the CLI setup project, which is written in typescript, having an available decorator.

How do we ensure that when we do a change in our "core" decorator, or that we add a new one, that new project thats gets generated with the CLI also benefit from them ?

And that an old project that want to update is able to get the new updated version of those ?

Reply:
> How do we ensure that when we do a change in our "core" decorator, or that we add a new one, that new project thats gets generated with the CLI also benefit from them ?

Perhaps the CLI should clone a package from our repo. That way, we can test it a bit and ensure that the CLI uses the correct version of Atomic for given files.

Reply:
We use and expose methods like `resultContext` & `initializeBindings` (perhaps more coming). The customer using the CLI would just get a simple decorator using these methods. As long as the api of stencil/core & our methods are still valid, the decorators should always work. 

As to were it comes from, we could have the decorators in the atomic package directly, and it would be copied in the CLI generated project source code. This is important because it has to use their stencil/core instance. It's also better to have it in the source code so the user can see & edit the decorators to his needs.